### ISSUE TITLE
Picking the worst path

### ISSUE THREAD

**Issue by RushTheOne** [Rikora/A-star#1]
The change to use priority_queue actually made the pathing move away from the target and make really bad paths.  Changing back to using a set fixed the issue.

**Comment by rkocjh1997** [comment]
Yes, I got the same case, The reason for this case is that `std::priority` is **Max Heap**. When the code run in loop, `m_openList.top().pos` will get the pos of max `f` instead of min `f`. SO, you should change `std::priority_queue<Node> m_openList;` to `std::priority_queue<Node, std::vector<Node>, std::greater<>> m_openList;`.
